substrate/client/network/sync/src/justification_requests.rs

```diff
@@ -117,6 +170,10 @@ impl<B: BlockT> ExtraRequests<B> {
 	pub(crate) fn peer_disconnected(&mut self, who: &PeerId) {
 		if let Some(request) = self.active_requests.remove(who) {
 			self.pending_requests.push_front(request);
+			if let Some(metrics) = &self.metrics {
+				metrics.active.dec();
+				metrics.pending.inc();
```

Comment:
Possible race condition (see the comment in `on_response()`), but the bug is already present without this PR.

Reply:
Yep, but it is narrow and only affects metrics, so :man_shrugging: 

I was thinking about changing these metrics to counters instead (so you not only know current state, but also number of transitions between metrics polling), but that would break metrics "interface", so I didn't change anything that affects the logic.